src/Suggestor/Suggestor.ts

```diff
@@ -317,10 +320,9 @@ function addRecurrenceSuggestions(line: string, cursorPos: number, settings: Set
  * Matches a string with a regex according to a position (typically of a cursor).
  * Will return a result only if a match exists and the given position is part of it.
  */
-export function matchByPosition(s: string, r: RegExp, position: number): RegExpMatchArray {
+export function matchByPosition(s: string, r: RegExp, position: number): RegExpMatchArray | void {
     const matches = s.matchAll(r);
     for (const match of matches) {
         if (match?.index && match.index <= position && position <= match.index + match[0].length) return match;
     }
-    return [];
```

Comment:
My text editor was correctly complaining about the empty array being returned not being a `RegExpMatchArray`. It also did not match the description saying it only returns a result when there is a true match, and every time `matchByPosition` was used, it was checking for the existence of the return value like so:
```ts
const wordMatch = matchByPosition( /*...*/ )'
if(wordMatch && wordMatch.length > 0) {
```
`wordMatch` is always truthy if it's an empty array or a `RegExpMatchArray`.

Reply:
Did you definitely run `yarn` before working on this?

I've only seen that message when my `node_modules` content is inconsistent with `yarn.lock`.

Reply:
Cancel that.

Reply:
Thank you for the reply.

There have been various errors in dependabot PRs updating typescript all around this area, which is why I was asking earlier if you environment didn't have yarn run.

However, not to worry, I am hoping that merging this PR will make them go away and enable me to accept more dependabot PRs.